### ISSUE TITLE
Detaching single tensors from computation graph

### ISSUE THREAD

**Issue by dominikj2** [mlverse/torch#596]
Hi, 

I’m aware of `with_no_grad` for detaching a sub-graph from a differentiable computation graph, but how do you detach a single variable/tensor?  Similar to` PyTorch .detach() `method

I tried to change the requires_grad flags and got the below error. The suggested `var.detach()` approach seems to allude to the `PyTorch .detach()` method.

```
Output$requires_grad <- FALSE
Error in (function (self, requires_grad)  : 
  you can only change requires_grad flags of leaf variables. If you want to use a computed variable in a subgraph that doesn't require differentiation use var_no_grad = var.detach().
```

Thanks.

**Comment by dfalbel** [comment]
You can do `x$detach()` or `x$detach_()` depending on if you want to create a copy or detach the tensor in-place. For example:

``` r
library(torch)
x <- torch_tensor(1, requires_grad = TRUE)
x_no_grad <- x$detach()
x_no_grad$requires_grad
#> [1] FALSE
```

**Comment by dominikj2** [comment]
Thank you. That is what I needed.

I regular perform key word searches from the reference list, [(https://torch.mlverse.org/docs/reference/index.html)](url), but it does not contain all the tensor methods available in the drop down menu for x$ (i.e. x$detach()). I can't seem to find a site (similar to Pytorch [https://pytorch.org/docs/stable/tensors.html](url)) that lists all available R Torch tensor methods? 
If one doesn't exist can you suggest how to extract the tensors drop down menu for easy key word searches.

Thanks again!

**Comment by dfalbel** [comment]
Cool! Glad it worked!

In theory they should be listed here: https://torch.mlverse.org/docs/articles/tensor/index.html
But docs can be in-complete depending on the method. 

If you use RStudio typing `x$` and then tab should show a list of all available methods too.

**Comment by dominikj2** [comment]
Great. That’s what I was after.

I wonder whether the below link should be in reference section and not in articles section:

https://torch.mlverse.org/docs/articles/tensor/index.html<https://torch.mlverse.org/docs/articles/tensor/index.html>

Thanks!

Dom

**Comment by dominikj2** [comment]
@dfalbel thanks for pointing me to: 
[https://torch.mlverse.org/docs/articles/tensor/index.html](url)

I wonder whether the link is better placed in reference section and not in articles section. 

Thanks!

**Comment by dfalbel** [comment]
Yes, definitely makes sense!
